[expo-updates] Add fingerprintExperimental runtime version policy (#24126)
expo/expo: docs/pages/eas-update/runtime-versions.mdx

Also, if you select a different native version between iOS and Android, you'll end up with builds and updates with separate runtime versions. When you publish an update, EAS CLI will detect that two updates are needed, and it will create two updates with the required runtime versions and publish them separately.

### `"fingerprint"` runtime version policy (experimental)

We provide the `"fingerprint"` runtime version policy for a project that has custom native code that may change between builds:

```json
{
  "expo": {
    "runtimeVersion": {
      "policy": "fingerprint"
    }
  }
}
```

> This policy is well-suited for projects incorporating custom native code that require safe updates without native conflicts. Unlike the "nativeVersion" policy, the fingerprint option automatically scans your project's native dependencies to update the runtimeVersion accordingly.

Reviewer: ```suggestion
This policy is well-suited for projects incorporating custom native code that require safe updates without native conflicts. Unlike the `"nativeVersion"` policy, the `"fingerprint"` option automatically scans your project's native dependencies to update the `"runtimeVersion"` accordingly.
```

Missing inline syntax highlights to match the consistency for this doc

> Please note that enabling fingerprinting may lengthen build times, as the fingerprinting process can be resource-intensive.

Reviewer: ```suggestion
Enabling fingerprinting may lengthen build times, as the fingerprinting process can be resource-intensive.
```
I appreciate maintaining positivity but we can avoid using words like "please".

> Attention: Use of the experimental fingerprint runtime policy may result in unexpected system behavior. This policy is still under active development as part of the expo-fingerprint beta package, and its specifications may change in the future.

Reviewer: ```suggestion
> **warning** Using the experimental fingerprint runtime policy may result in unexpected system behavior. This policy is still under active development as part of the `expo-fingerprint` beta package, and its specifications may change in the future.
```

If this needs to be a note or a warning, let's use Blockquote (`>`) or a warning callout (`> **warning**`). I personally think since this is in beta, we can add a warning callout since that's the pattern we follow in Expo docs.
